.github/workflows/go.yml

```diff
@@ -0,0 +1,25 @@
+name: Go
```

Comment:
You might want to give it more generic name, e.g. `Build`. At the end you'll not only compile Go, but also lint, run unit tests, and submit codecov reports.